src/methods/join.js

```diff
@@ -588,6 +588,18 @@ export class JoinImplementation {
     return [result, generator, bindings, properties, createdObjects];
   }
 
+  joinNestedEffects(realm: Realm, c: Completion | Value, precedingEffects?: Effects): Effects {
```

Comment:
The precedingEffects arguments seems difficult to use from the outside since it is highly dependent upon what is in the c argument. Perhaps, it should be private and the public API shouldn't allow this argument.